aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ResourceModelHelper.java

```diff
@@ -11,9 +11,12 @@
 import java.util.Optional;
 
 public final class ResourceModelHelper {
-    private final static String STORAGE_TYPE_IO1 = "io1";
+    private static final List<String> STORAGE_TYPES_SET_ON_MODIFY_AFTER_CREATE = Arrays.asList(
+            "io1",
+            "gp3"
+    );
```

Comment:
I'd say: go for an Enum for storage types and a Set for `STORAGE_TYPES_SET_ON_MODIFY_AFTER_CREATE`